sound/soc/sof/core.c

```diff
@@ -478,9 +467,9 @@ int snd_sof_device_probe(struct platform_device *pdev)
 	/* initialize sof device */
 	sdev->dev = &pdev->dev;
 	sdev->pci = to_pci_dev(pdev->dev.parent);
+	sdev->sof_mdev = dev_get_drvdata(pdev->dev.parent);
```

Comment:
this confuses me quite a bit.

It looks like the IPC moves to the PCI device level then? Well, what about all the hardware initialization then? All the controller hardware initialization stuff is now handled in the platform device when it probes, so the partitioning looks very odd to me.

Wondering if we should instead split the core and instead of creating the MFD children at the top level do it in the platform specific level, on a need-basis if supported, and then have the children use parts of the existing SOF core to complete e.g. topology load and card creation.